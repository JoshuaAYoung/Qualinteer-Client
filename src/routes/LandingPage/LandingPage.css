.landingBanner {
  margin: -8rem -2.5rem 5rem;
  padding: 8rem 2.5rem 0;
  height: 50rem;
  border-bottom: .1rem solid black;
  background: #FFF0B0;
  position: relative;
}

.heroImage {
  width: 30.4rem;
  position: absolute;
  bottom: 0;
  right: 5%;
}

.actionImage {
  width: 16rem;
}

.actionVolContainer {
  margin-bottom: 6rem;
}

.featureSubHeader {
  padding-left: calc(50% - 16rem);
  margin-top: 0;
  position: relative;
  padding-top: 3rem;
}

.actionInstructions {
  padding: 0 2rem;
  font-size: 1.4rem;
  margin-bottom: 2rem;
}

.actionInstructionsVol {
  padding-top: 2rem;
}

.actionInstructionsOrg {
  padding-top: 2rem;
}

.actionLink {
  color: black;
  text-decoration: none;
}

.callToAction {
  margin-bottom: 8rem;
}

.appFeatures {
  margin: 0 -2.5rem;
  position: relative;
}

.bottomWaves {
  height: 2rem;
}

.topWaves {
  height: 2rem;
  object-position: -16rem;
}

.waveContainer {
  overflow: hidden;
  height: 2rem;
}

.featuresContainer {
  padding: 0 2.5rem;
  background: #FFF0B0;
}

.handsImage {
  width: 14rem;
  position: absolute;
  top: -7rem;
  right: calc(50% - 11rem);
}

.features {
  display: inline-block;
}

.feature {
  width: calc(100vw - 10.5rem);
  display: inline-block;
  margin-bottom: 2.4rem;
}

.featureCheck {
  width: 3rem;
  margin-right: 1.5rem;
  vertical-align: 1rem;
}


.megaphoneImage {
  width: 8rem;
  display: inline-block;
  margin: 0 1.2rem 0 -2rem;
  vertical-align: -1rem;
}

.testimonialQuote {
  font-style: italic;
  margin-bottom: 1.5rem;
}

.testimonialAuthor {
  margin-bottom: 3.5rem;
}

.testimonials {
  margin-top: 2rem;
}


.featuresSubHeader {
  margin-top: 0;
  }

@media only screen and (max-width: 360px) {
  .heroImage {
    width: 27rem;
  }

  .handsImage {
    width: 12rem;
    right: calc(50% - 6rem); 
  }

  .featuresSubHeader {
  margin-top: 3rem;
  text-align: center;
  margin-bottom: 4rem;
  }
}

@media only screen and (min-width: 1024px) {
  .landingBanner {
    padding-top: 19rem;
  }
  .heroImage {
    width: 43rem;
    right: calc(49vw - 42rem);
  }
  .heroTextContainer {
    left: calc(49vw - 42rem);
    width: 42rem;
    position: absolute;
  }
  .landingBanner{
    margin-right: calc(42rem - 49vw);
    margin-left: calc(42rem - 49vw);
  }

  .actionVolContainer {
    text-align: right;
    margin-right: 6rem;
  }

  .actionOrgContainer {
    text-align: left;
    margin-left: 6rem;
  }

  .actionSeparator {
    border-right: .2rem solid #F4BD5B;
    height: 50rem;
    align-self: center
  }

  .callToAction {
    display: flex;
    margin-bottom: 4rem;
  }

  .actionInstructionsVol {
    padding-left: 5rem;
  }

  .actionInstructionsOrg {
    padding-right: 5rem;
  }

  .actionInstructions {
    padding: 0;
  }

  .handsImage {
    top: -7rem;
  }

  .features {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    padding-bottom: 3rem;
  }

  .featuresContainer {
    padding-right: calc(49vw - 33rem);
    padding-left: calc(49vw - 33rem);
  }

  .featureContainer {
    width: 31rem
  }

  .feature {
    width: 26rem;
  }
  
  .featureCheck {
    vertical-align: 3rem;
  }

  .featureSubHeader {
    margin-left: calc(50% - 16rem);
    margin-bottom: 6rem;
    width: 32rem;
    padding-top: 3rem;
    padding-left: 0;
  }

  .testimonialsContainer {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: flex-start;
  }

  .testimonial {
    width: 46%;
  }

  .testimonials {
    margin: 4rem 3rem 0;
  }

}