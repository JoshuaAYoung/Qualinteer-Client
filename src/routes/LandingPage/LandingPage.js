import React from 'react';

function LandingPage() {
  return (
    <div className="landingMain">
      <header role="banner" className="landingBanner">
        <h1>Qualinteer</h1>
        <h2>Gift your experience</h2>
      </header>
      <section className="landingDescription">
        <header>
          <h3>A Unique Way to Give Back</h3>
        </header>
        <p>Qualinteer is a place where non-profit organizations in need of skilled labor are connected with volunteers that have the exertise required to help.</p>
        <p>
          A lot of volunteer work consists of unskilled labor, which is great if that's what's needed. When it comes to shopping for professional services and skilled labor, though, non-profits tend to pay market rate.
        </p>
        <p>
          In part, this is an issue with the tools available. Hiring volunteers to pack boxes or serve soup is a simple process. Volunteer coordinators can blanket the social media airwaves with the same ad, since just about any warm body will do. Finding someone with the expertise to build you a website or design your new office can be a little more complicated, though. Also want them to work pro-bono? For that you need Qualinteer...
        </p>
      </section>
      <section className="landingInstructions">
        <header>
          <h3>Let's Get Giving</h3>
        </header>
        <p>If you want to volunteer, take a look at the job board (no login required) to find ways in which you can give back. Whether you're seeking help or offering it, you can click on the "Sign up" button at the top of the page to create an account. If you already have an account, you can click on "Log in" in the navigation bar.</p>
        <p>If you specify that you're looking to volunteer, once you login you'll be able to apply to any of the jobs listed. If you specify that you're in need of volunteers, once you log in you'll be able to post a job to the job board.</p>
      </section>
    </div>
  );
}

export default LandingPage;