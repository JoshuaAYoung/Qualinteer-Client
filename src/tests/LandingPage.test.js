import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LandingPage from '../routes/LandingPage/LandingPage';

describe('LandingPage Component', () => {
  it('LandingPage Smoke Test', () => {
    const div = document.createElement('div');

    ReactDOM.render(<LandingPage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('LandingPage Snapshot Test', () => {
    const wrapper = shallow(<LandingPage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
