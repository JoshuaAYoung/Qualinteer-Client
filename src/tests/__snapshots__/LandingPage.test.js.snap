// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component LandingPage Snapshot Test 1`] = `
<div
  className="landingMain"
>
  <div
    className="landingBanner"
  >
    <div
      className="heroTextContainer"
    >
      <h1
        className="pageHeader"
      >
        Experience as a Gift.
      </h1>
      <p
        className="pageInstructions landingInstructions"
      >
        <span
          className="defaultMedium"
        >
          v
          <span
            className="inlineBullet"
          >
            •
          </span>
          able
        </span>
         is a platform for connecting 
        <span
          className="defaultMedium"
        >
          <u>
            viable
          </u>
        </span>
        , 
        <span
          className="defaultMedium"
        >
          <u>
            volunteerable
          </u>
        </span>
        , 
        <span
          className="defaultMedium"
        >
          <u>
            very able
          </u>
        </span>
         people to the causes that they find most 
        <span
          className="defaultMedium"
        >
          <u>
            valuable
          </u>
        </span>
        .
      </p>
    </div>
    <img
      alt="people working on a spaceship"
      className="heroImage"
      src="/assets/landing-hero.svg"
    />
  </div>
  <section
    className="callToAction textCenter"
  >
    <div
      className="actionVolContainer"
    >
      <div
        className="actionDescriptionContainer"
      >
        <h2
          className="subHeader"
        >
          Offer Your Expertise.
        </h2>
        <p>
          Are you a volunteer who’s passionate about giving back? v
          <span
            className="inlineBullet"
          >
            •
          </span>
          able makes it easy to find causes that inspire you. And by offering up your experience in addition to your time, you’re increasing the value of every single minute spent volunteering.
        </p>
      </div>
      <div
        className="actionInstructionsVol"
      >
        <img
          alt="a computer with an application"
          className="actionImage"
          src="/assets/volunteer.svg"
        />
        <p
          className="actionInstructions"
        >
          Click on the “Volunteer” button to be directed to the opportunity board page to see available volunteer positions and to apply.
        </p>
        <button
          className="actionButton"
          onClick={[Function]}
        >
          Volunteer
        </button>
      </div>
    </div>
    <div
      className="actionSeparator"
    />
    <div
      className="actionOrgContainer"
    >
      <div
        className="actionDescriptionContainer"
      >
        <h2
          className="subHeader"
        >
          Find skilled help.
        </h2>
        <p>
          As a non-profit organization, we know how hard it can be to find passionate volunteers. Looking for a passionate volunteer who can also build you a website or design you a new office? For that you need an extensive network of skilled individuals like v
          <span
            className="inlineBullet"
          >
            •
          </span>
          able.
        </p>
      </div>
      <div
        className="actionInstructionsOrg"
      >
        <img
          alt="a woman connecting with people"
          className="actionImage"
          src="/assets/recruit.svg"
        />
        <p
          className="actionInstructions"
        >
          Once you’ve logged in as an organization, click the “Recruit Help” button to post  an opportunity and find volunteers.
        </p>
        <button
          className="actionButton"
          onClick={[Function]}
        >
          Recruit Help
        </button>
      </div>
    </div>
  </section>
  <section
    className="appFeatures"
  >
    <div
      className="waveContainer"
    >
      <img
        alt="wavy border"
        className="topWaves"
        src="/assets/top-waves.svg"
      />
    </div>
    <div
      className="featuresContainer"
    >
      <div
        className="featureSubHeader"
      >
        <h2
          className="subHeader featuresSubHeader"
        >
          Now this is
          <br />
          feature-rich volunteering.
        </h2>
        <img
          alt="hands up holding hearts"
          className="handsImage"
          src="/assets/hands-hearts.svg"
        />
      </div>
      <div
        className="features"
      >
        <div
          className="featureContainer"
        >
          <img
            alt="checkmark"
            className="featureCheck"
            src="/assets/checkmark.svg"
          />
          <div
            className="feature"
          >
            <h4
              className="smallHeader"
            >
              Opportunity Board
            </h4>
            <p>
              All opportunities in one place. Search by location or keyword.
            </p>
          </div>
        </div>
        <div
          className="featureContainer"
        >
          <img
            alt="checkmark"
            className="featureCheck"
            src="/assets/checkmark.svg"
          />
          <div
            className="feature"
          >
            <h4
              className="smallHeader"
            >
              Registration and Login
            </h4>
            <p>
              Real authentication using advanced security techniques.
            </p>
          </div>
        </div>
        <div
          className="featureContainer"
        >
          <img
            alt="checkmark"
            className="featureCheck"
            src="/assets/checkmark.svg"
          />
          <div
            className="feature"
          >
            <h4
              className="smallHeader"
            >
              Opportunity Post
            </h4>
            <p>
              Create an opportunity using the embedded rich text editor.
            </p>
          </div>
        </div>
        <div
          className="featureContainer"
        >
          <img
            alt="checkmark"
            className="featureCheck"
            src="/assets/checkmark.svg"
          />
          <div
            className="feature"
          >
            <h4
              className="smallHeader"
            >
              Volunteer Button
            </h4>
            <p>
              Opens the default email client with pre-populated info.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="waveContainer"
    >
      <img
        alt="wavy border"
        className="bottomWaves"
        src="/assets/bottom-waves.svg"
      />
    </div>
  </section>
  <section
    className="testimonials textCenter"
  >
    <img
      alt="megaphone"
      className="megaphoneImage"
      src="/assets/megaphone.svg"
    />
    <h2
      className="subHeader inline"
    >
      Testimonials.
    </h2>
    <div
      className="testimonialsContainer"
    >
      <div
        className="testimonial"
      >
        <p
          className="testimonialQuote"
        >
          “We had no problem getting volunteers to pack trucks. But when we needed an architect, we had a difficult time finding one that would work pro-bono.  That is, until we discovered v
          <span
            className="inlineBullet"
          >
            •
          </span>
          able.”
        </p>
        <p
          className="testimonialAuthor defaultMedium"
        >
          - Rachael Goodman (Dance for Humanity)
        </p>
      </div>
      <div
        className="testimonial"
      >
        <p
          className="testimonialQuote"
        >
          “We made a lot of calls trying to find a qualified volunteer veterinarian with no luck. A friend told us to check out v•able and the rest is history. Dr. Lolanda has been with us for 4 years now and we couldn’t be happier.”
        </p>
        <p
          className="testimonialAuthor defaultMedium"
        >
          - Broderick Hart (All Cats Foundation)
        </p>
      </div>
      <div
        className="testimonial"
      >
        <p
          className="testimonialQuote"
        >
          “We’ve used v•able for years for all of our volunteer needs. It’s a perfect fit for us.”
        </p>
        <p
          className="testimonialAuthor defaultMedium"
        >
          - Sadie Giver (The Bunny Sanctuary)
        </p>
      </div>
      <div
        className="testimonial"
      >
        <p
          className="testimonialQuote"
        >
          “v
          <span
            className="inlineBullet"
          >
            •
          </span>
          able makes finding skilled volunteers so simple and easy. I wish I had found it sooner!”
        </p>
        <p
          className="testimonialAuthor defaultMedium"
        >
          - Miranda Sainthouse (Burn Awareness Institute)
        </p>
      </div>
    </div>
  </section>
</div>
`;
